src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -109,97 +137,77 @@ public Task<FeatureDefinition> GetFeatureDefinitionAsync(string featureName)
+            IEnumerable<IConfigurationSection> dotnetFeatureManagementSection = GetDotnetFeatureManagementSection(rootConfigurationFallback);
 
-                if (MicrosoftFeatureManagementConfigurationSection == null & RootConfigurationFallbackEnabled)
+            if (dotnetFeatureManagementSection != null)
+            {
+                foreach (IConfigurationSection featureSection in dotnetFeatureManagementSection)
                 {
-                    IConfiguration featureFlagsSection = _configuration
-                        .GetChildren()
-                        .FirstOrDefault(section =>
-                            string.Equals(
-                                section.Key,
-                                MicrosoftFeatureManagementFields.FeatureFlagsSectionName,
-                                StringComparison.OrdinalIgnoreCase));
-
-                    hasMicrosoftFeatureManagementSchema = featureFlagsSection != null;
-                }
+                    string featureName = featureSection.Key;
 
-                lock (_lock)
-                {
-                    if (Interlocked.Read(ref _initialized) == 0)
+                    if (string.IsNullOrEmpty(featureName))
                     {
-                        _microsoftFeatureManagementSchemaEnabled = hasMicrosoftFeatureManagementSchema;
+                        continue;
+                    }
+
+                    //
+                    // Underlying IConfigurationSection data is dynamic so latest feature definitions are returned
+                    FeatureDefinition definition = _definitions.GetOrAdd(featureName, (_) => ParseDotnetFeatureDefinition(featureSection));
 
-                        Interlocked.Exchange(ref _initialized, 1);
+                    if (definition != null)
+                    {
+                        yield return definition;
                     }
                 }
             }
         }
 
-        private FeatureDefinition ReadFeatureDefinition(string featureName)
+        private IEnumerable<IConfigurationSection> GetDotnetFeatureManagementSection(bool rootConfigurationFallback)
         {
-            IConfigurationSection configuration = GetFeatureDefinitionSections()
-                .FirstOrDefault(section => string.Equals(GetFeatureName(section), featureName, StringComparison.OrdinalIgnoreCase));
+            IEnumerable<IConfigurationSection> dotnetFeatureManagementSection = GetFeatureDefinitionSections(DotnetFeatureManagementFields.FeatureManagementSectionName);
 
-            if (configuration == null)
+            if (dotnetFeatureManagementSection == null && rootConfigurationFallback)
             {
-                return null;
-            }
-
-            return ReadFeatureDefinition(configuration);
-        }
+                if (!_configuration.GetChildren().Any())
+                {
+                    Logger?.LogDebug($"Configuration is empty.");
```

Comment:
Who is this message intended for?

Reply:
It used to be logged when you passed an empty configuration to `AddFeatureManagement(configuration)`.